lib/galaxy/jobs/runners/kubernetes.py

```diff
@@ -271,6 +283,99 @@ def __get_k8s_job_spec_template(self, ajs):
         k8s_spec_template["metadata"]["annotations"].update(extra_metadata.get('annotations', {}))
         return k8s_spec_template
 
+    def __get_k8s_service_spec(self, ajs):
+        """The k8s spec template is nothing but a Service spec, except that it is nested and does not have an apiversion
+        nor kind."""
+        container = self._find_container(ajs.job_wrapper)
+        guest_ports = container.tool_info.guest_ports
+        if not guest_ports:
+            return None
+        else:
+            k8s_spec_template = {
+                "metadata": {
+                    "labels": {
+                        "app.galaxyproject.org/handler": self.__force_label_conformity(self.app.config.server_name),
+                        "app.galaxyproject.org/destination": self.__force_label_conformity(
+                            str(ajs.job_wrapper.job_destination.id))
+                    },
+                    "annotations": {
+                        "app.galaxyproject.org/tool_id": ajs.job_wrapper.tool.id
+                    }
+                },
+                "spec": {
+                    "ports": [{"name": "job-{}-{}".format(self.__force_label_conformity(ajs.job_wrapper.get_id_tag()), p),
+                               "port": int(p),
+                               "protocol": "TCP",
+                               "targetPort": int(p)} for p in guest_ports],
+                    "selector": {
+                        "app.kubernetes.io/name": self.__force_label_conformity(ajs.job_wrapper.tool.old_id),
+                        "app.kubernetes.io/component": "tool",
+                        "app.galaxyproject.org/job_id": self.__force_label_conformity(ajs.job_wrapper.get_id_tag())
+                    },
+                    "type": "ClusterIP"
+                }
+            }
+            log.debug(k8s_spec_template)
```

Comment:
Same here, if you don't need it, to drop the debug, otherwise prefix this with a message that says what this is.